leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/CaliforniumLwM2mRequestSender.java

```diff
@@ -80,34 +83,57 @@ public CaliforniumLwM2mRequestSender(final Set<Endpoint> endpoints, final Observ
             public T buildResponse(final Response coapResponse) {
                 // Build LwM2m response
                 final LwM2mResponseBuilder<T> lwm2mResponseBuilder = new LwM2mResponseBuilder<T>(coapRequest,
-                        coapResponse, destination, model, observationRegistry);
+                        coapResponse, registrationInfo, model, observationRegistry);
                 request.accept(lwm2mResponseBuilder);
                 return lwm2mResponseBuilder.getResponse();
             }
         };
         coapRequest.addMessageObserver(syncMessageObserver);
 
-        // Store pending request to cancel it on deregistration
-        addPendingRequest(destination.getRegistrationId(), coapRequest);
+        // Store pending request to cancel it on de-registration
+        addPendingRequest(registrationInfo.getRegistrationId(), coapRequest);
 
         // Send CoAP request asynchronously
-        final Endpoint endpoint = getEndpointForClient(destination);
+        final Endpoint endpoint = getEndpointForClient(registrationInfo);
         endpoint.sendRequest(coapRequest);
 
         // Wait for response, then return it
         return syncMessageObserver.waitForResponse();
     }
 
     @Override
-    public <T extends LwM2mResponse> void send(final Client destination, final DownlinkRequest<T> request,
+    public <T extends LwM2mResponse> void send(final Client registrationInfo, final DownlinkRequest<T> request,
             final ResponseCallback<T> responseCallback, final ErrorCallback errorCallback) {
+        ayncSendUsingCoapEndpoint(registrationInfo, request, responseCallback, errorCallback);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.eclipse.leshan.server.request.LwM2mRequestSender#send(org.eclipse.leshan.server.client.Client,
+     * java.lang.String, org.eclipse.leshan.core.request.DownlinkRequest)
+     */
+    @SuppressWarnings("unchecked") // Java generic usage and strict type has to be removed from Leshan
+    @Override
+    public <T extends LwM2mResponse> void send(final Client registrationInfo, final String requestTicket,
+            final DownlinkRequest<T> request) {
```

Comment:
Currently in Leshan, we get notifications through the `ObservationRegistry`. So no ticket for now. We will see later if we remove ObservationRegistry and get all notification through `ResponseListener` API. (One step at a time)